services/webnn/ort/graph_builder_ort.cc

```diff
@@ -1031,92 +1031,90 @@ GraphBuilderOrt::AddDequantizeLinearOperation(
   std::vector<uint32_t> scale_shape = scale_descriptor.shape();
 
   int64_t axis = 0;
-  int64_t block_size;
-  bool need_transpose = false;
-
-  uint32_t not_one_value_count = 0;
-  bool is_per_axis = false;
+  uint32_t not_one_value_dim_count = 0;
+  bool found_same_size = false;
   CHECK_LE(scale_shape.size(), input_shape.size());
   for (size_t i = 0; i < scale_shape.size(); i++) {
     if (scale_shape[scale_shape.size() - i - 1] != 1) {
-      not_one_value_count++;
-      if (not_one_value_count == 1 &&
-          scale_shape[scale_shape.size() - i - 1] ==
-              input_shape[input_shape.size() - i - 1]) {
+      not_one_value_dim_count++;
+      if (scale_shape[scale_shape.size() - i - 1] ==
+          input_shape[input_shape.size() - i - 1]) {
         axis = input_shape.size() - i - 1;
-        is_per_axis = true;
+        found_same_size = true;
       }
     }
   }
+  // TODO(https://github.com/shiyi9801/chromium/issues/139): Consider to add
+  // emulation to support multiple axes case, e.g. input shape is [2, 3, 4, 5]
+  // and scale shape is [1, 3, 4, 1].
+  bool is_per_axis = found_same_size && not_one_value_dim_count == 1;
 
-  // Each value of scale_shape is 1.
-  if (not_one_value_count == 0) {
-    auto it = std::find(input_shape.begin(), input_shape.end(), 1);
-    if (it != input_shape.end()) {
-      axis = std::distance(input_shape.begin(), it);
-      is_per_axis = true;
-    }
-  }
-
+  int64_t block_size;
+  bool need_transpose = false;
   if (scale_shape.empty()) {
     // For per-tensor/layer dequantization the scale is a scalar.
     axis = 0;
     // block_size must be 0 for per-tensor quantization.
     block_size = 0;
-  } else if (not_one_value_count <= 1 && is_per_axis) {
-    // for per per-axis dequantization, scale and zeroPoint must be a 1-D
+  } else if (not_one_value_dim_count == 0) {
+    // The numbers in scale shape are all 1., scale and zeroPoint should be
+    // reshaped to a scalar.
+    ASSIGN_OR_RETURN(scale_name, PrependReshape(scale_name, {}));
+    ASSIGN_OR_RETURN(zero_point_name, PrependReshape(zero_point_name, {}));
+    axis = 0;
+    // block_size must be 0 for per-tensor quantization.
+    block_size = 0;
```

Comment:
Same here

Reply:
If block_wise is not set to 0, ORT will throw an error for per axis dequantize- "input rank != scale_rank". I suppose ORT implementation check whether the case is block_wise according to the value of `block_size`.